### PR TITLE
Make certificates validity configurable via EtcdStorage Spec and define default values

### LINKED ISSUE #52: Make certificates validity configurable
We have established the following defaults for certificates validity:
* 5 years for CA certificates,
* 3 years for server certificates,
* 1 month for client certificates.

This should be configurable by users (EtcdProxyController operators).

We should also add limits for minimal validity, which should be 5 minutes.

### PR DESCRIPTION
**Pull request by xmudrii**
Closes #52 

This PR adds three new fields to the EtcdStorage Spec for defining etcd-proxy certificate validity:
* `signingCertificateValidity` - duration for how long the self-signed singing certificate is valid. Takes duration in form of a string as defined [here (GoDoc)](https://golang.org/pkg/time/#ParseDuration).
* `servingCertificateValidity` - duration for how long the serving certificate/key pair is valid. Takes duration in form of a string as defined [here (GoDoc)](https://golang.org/pkg/time/#ParseDuration)..
* `clientCertificateValidity` - duration for how long the client certificate/key pair is valid. Takes duration in form of a string as defined [here (GoDoc)](https://golang.org/pkg/time/#ParseDuration).

As there is no CRD Defaulting in place for now, the operator must provide those three fields when creating EtcdStroage resource.

### CODE REVIEW

File: artifacts/deployment/00-etcdproxy-controller.yaml [hunk 1]
```diff
@@ -127,6 +127,18 @@ spec:
                   namespace:
                     type: string
                     pattern: '^[a-z0-9]([-a-z0-9]*[a-z0-9])?$'
+            signingCertificateValidityMinutes:
```
Comment (sttts) [inline artifacts/deployment/00-etcdproxy-controller.yaml]: required?

File: artifacts/etcdstorage/crd.yaml [hunk 2]
```diff
@@ -44,4 +44,16 @@ spec:
...
+            signingCertificateValidityMinutes:
+              type: integer
+              minimum: 10
+              maximum: 43830
+            servingCertificateValidityMinutes:
+              type: integer
+              minimum: 10
+              maximum: 43830
+            clientCertificateValidityMinutes:
+              type: integer
```
Comment (sttts) [inline artifacts/etcdstorage/crd.yaml]: we also have a `time.Duration` wrapper in apimachinery. Then you can write 10m and 0.5y.

File: artifacts/deployment/00-etcdproxy-controller.yaml (hunk 1 above)
Comment (xmudrii) [reply artifacts/deployment/00-etcdproxy-controller.yaml]: Done.

**Comment by deads2k** [comment]
feel free to merge on green.

**Pull request merged**